### PR TITLE
GAP: added option to set filter accept list

### PR DESCRIPTION
**Pull request by jrotkiewicz**
Dialog board sends 0 after reset. Added fix to discard this zero.
Divided PR: https://github.com/intel/auto-pts/pull/585

### CODE REVIEW

**Comment by jrotkiewicz** [comment]
@michal-narajowski

File: doc/btp_spec.txt [hunk 1]
```diff
@@ -383,6 +383,8 @@ Commands and responses:
 
 		This command is used to create a Link Layer connection with
 		remote device.
+		Address 00:00:00:00:00:00 will attempt to use address
```
Comment (michal-narajowski) [inline doc/btp_spec.txt:386]: Address should be 00:00:00:00:00:00 and address type should be 0

File: doc/btp_spec.txt [hunk 2]
```diff
@@ -567,6 +569,21 @@ Commands and responses:
 
 		In case of an error, the error response will be returned.
 
+	Opcode 0x1f - Set filter accept list
+
+		Command parameters:
```
Comment (michal-narajowski) [inline doc/btp_spec.txt:574]:
```suggestion
		Controller Index: <controller id>
		Command parameters:
```

File: doc/btp_spec.txt [hunk 3]
```diff
@@ -567,6 +569,21 @@ Commands and responses:
 
 		In case of an error, the error response will be returned.
 
+	Opcode 0x1f - Set filter accept list
```
Comment (michal-narajowski) [inline doc/btp_spec.txt:572]:
```suggestion
	Opcode 0x1c - Set filter accept list
```

File: doc/btp_spec.txt [hunk 4]
```diff
@@ -567,6 +569,21 @@ Commands and responses:
...
+
+		Command parameters:
+			Address tuples:
+				Address_Type (1 octet)
+				Address (6 octets)
+			Example:
+				addr_tup = [('11:22:33:44:55:66', 0), ...]
+
+		This command is used to set filter accept list.
+		When set, connect with address: 00:00:00:00:00:00
```
Comment (michal-narajowski) [inline doc/btp_spec.txt:582]: address 0 and type 0.

File: doc/btp_spec.txt [hunk 5]
```diff
@@ -567,6 +569,21 @@ Commands and responses:
...
 		In case of an error, the error response will be returned.
 
+	Opcode 0x1f - Set filter accept list
+
+		Command parameters:
+			Address tuples:
+				Address_Type (1 octet)
+				Address (6 octets)
+			Example:
+				addr_tup = [('11:22:33:44:55:66', 0), ...]
```
Comment (michal-narajowski) [inline doc/btp_spec.txt:579]:
```suggestion
```

File: doc/btp_spec.txt [hunk 6]
```diff
@@ -567,6 +569,21 @@ Commands and responses:
 
 		In case of an error, the error response will be returned.
 
+	Opcode 0x1f - Set filter accept list
+
+		Command parameters:
+			Address tuples:
```
Comment (michal-narajowski) [inline doc/btp_spec.txt:575]:
```suggestion
			Address_Type #N (1 octet)
			Address #N (6 octets)
			for N from 1 to the number of addresses in the filter accept list.
```

File: pybtp/btp/gap.py [hunk 7]
```diff
@@ -439,6 +441,43 @@ def gap_conn(bd_addr=None, bd_addr_type=None, own_addr_type=OwnAddrType.le_ident
...
+        Arguments:
+        addr_tup -- addrs and their types as tuples:
+            addr_tup = [('DB:F5:72:56:C9:EF', 0), ('DB:F5:72:56:C9:EF', 0)]
+    """
+    logging.debug("%s %s", set_filter_accept_list.__name__, addr_tup)
+    iutctl = get_iut()
+
+    data_ba = bytearray()
+
+    if addr_tup == None:
```
Comment (michal-narajowski) [inline pybtp/btp/gap.py:458]:
```suggestion
    if not address_list:
        address_list = [(pts_addr_get(None), pts_addr_type_get(None))]
    
    for addr, type in address_list:
            bd_addr_ba = addr2btp_ba(addr)
            bd_addr_type_ba = chr(type).encode('utf-8')
            data_ba.extend(bd_addr_type_ba)
            data_ba.extend(bd_addr_ba)
    
    
```

File: pybtp/btp/gap.py [hunk 8]
```diff
@@ -439,6 +441,43 @@ def gap_conn(bd_addr=None, bd_addr_type=None, own_addr_type=OwnAddrType.le_ident
     gap_command_rsp_succ()
 
 
+def set_filter_accept_list(addr_tup=None):
```
Comment (michal-narajowski) [inline pybtp/btp/gap.py:444]:
```suggestion
def set_filter_accept_list(address_list=None):
```

**Review by michal-narajowski — CHANGES_REQUESTED** [review]

**Comment by jrotkiewicz** [comment]
@michal-narajowski

**Review by michal-narajowski — APPROVED** [review]

**Pull request merged**